### PR TITLE
Emit correct DWARF version

### PR DESCRIPTION
**Pull request by Luhrel**
First part of the implementation of DWARFv5.

### CODE REVIEW

File: src/dmd/backend/dwarf.d [hunk 1]
```diff
@@ -16,7 +16,7 @@ extern (C++):
 
 nothrow:
 
-enum DWARF_VERSION = 3;
+enum DWARF_VERSION = 5;
```
Comment (ibuclaw) [inline src/dmd/backend/dwarf.d:19]: You probably want to keep this to 4 for the time being, then allow emitting version 5 code as experimental.

**Comment by ibuclaw** [comment]
Are you going to send patches to gdb to make use of any of these features?

**Comment by Luhrel** [comment]
> Are you going to send patches to gdb to make use of any of these features?
No.
I will only implement new attributes and types.

File: src/dmd/backend/dwarf.d (hunk 1 above)
Comment (Luhrel) [reply src/dmd/backend/dwarf.d:19]: Should I make a compiler switch ?
Something like `--dwarf=5` ?

**Comment by Luhrel** [comment]
Ping @RazvanN7

**Comment by 12345swordy** [comment]
@luhrel try emailing @RazvanN7 for a PR review.

**Comment by RazvanN7** [comment]
The diff currently makes it hard to review this patch. It reads like a mix of refactoring and added code. Can you please try to minimize the diff by avoiding to change the indentation of some lines of code?

**Comment by Luhrel** [comment]
The first commit simply embraces the sections for proper reading, but yeah the diff quite s*ck.
I will redo my commits later on.

**Comment by RazvanN7** [comment]
PR looks ok, but I would rather have it split in 2: 1 that does emit the correct version (what the title of this PR says) and 1 that does the refactoring. This is the policy enforced by Walter, see [1] for reference.

[1] https://github.com/dlang/dmd/pull/12168

**Comment by Luhrel** [comment]
@RazvanN7 https://github.com/dlang/dmd/pull/12196

**Review by RazvanN7 — APPROVED** [review]
Thanks!

File: test/dshell/dwarf.d [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import dshell;
 
 import std.algorithm;
+import std.array;
```
Comment (RazvanN7) [inline test/dshell/dwarf.d:4]: Why is this import needed?

Comment (Luhrel) [reply test/dshell/dwarf.d]: it's a vestige, removed.
